lib/using_restricted_array.rb

```diff
@@ -1,61 +1,130 @@
+# frozen_string_literal: true
 require_relative 'restricted_array.rb'
+Minitest::Reporters.use!
+Minitest::Reporters::SpecReporter.new
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: 0(n) Linear: iteration depends on length, we have to go through the array until we find the termination character nil.
+# Space complexity: O(1); uses constant memory, doesn't matter how big array is, we have one counter, no new data structures being created
 def length(array)
-  raise NotImplementedError
+  # raise NotImplementedError
+  i = 0
+  i += 1 until array[i].nil?
+  i
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) linear, in order to print every element in array we have to go through entire array.
+# Space complexity: O(1) uses constant amount of memory, prints elements in array. No matter how array changes,
+# new memory is not taken up for this method.
 def print_array(array)
-  raise NotImplementedError
+  i = 0
+  until array[i].nil?
+    print array[i]
+    i += 1
+  end
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) linear, have to iterate through array until we find it.
+# Space complexity: O(1), constant memory,
 def search(array, length, value_to_find)
-  raise NotImplementedError
+  # raise NotImplementedError
+  length.times do |i|
+    return true if value_to_find == array[i]
+  end
+  false
```

Comment:
Just for readability I encourage the use of the `return` keyword

```suggestion
  return false
```